src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -105,7 +105,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
   // that are turned private by typedBlock
   private final val SYNTHETIC_PRIVATE = TRANS_FLAG
 
-  private final val InterpolatorCodeRegex  = """\$\{.*?\}""".r
+  private final val InterpolatorCodeRegex  = """\$\{(.*?)\}""".r
```

Comment:
Could we simplify `.*?` to `.*`

i won't add a ? at the end of the above question to avoid confusion..  :)